api/providers/AddressProvider.cs

```diff
@@ -53,4 +55,23 @@ public List<Address> GetAll()
     {
         throw new NotImplementedException();
     }
+
+    public Address? GetOrCreateAddress(SupplierRequest request)
+    {
+
+        if (request.AddressId != null)
+        {
+            Address? existingAddress = GetById(request.AddressId.Value);
+            if (existingAddress == null) throw new ApiFlowException("address_id does not exist");
+            return existingAddress;
+        }
+        Address address = Create<AddressDTO>(request.Address);
+
+        if (request.Address != null)
+        {
+            return address;
+        }
+
+        throw new ApiFlowException("Both address_id and address data are missing. Unable to create supplier address.");
+    }
```

Comment:
Deze functie is niet verantwoordelijk voor het valideren dat het address is gemaakt. Dit hoort gedaan te worden door de Validator. Deze functie is ook niet herbruikbaar omdat dit gebruik maakt van SupplierRequest dit moet naar AddressDTO. Hij moet puur een address aanmaken of zoeken anders null returnen.

```suggestion
    public Address? GetOrCreateAddress(AddressDTO? addressDTO = null, Guid? addressId = null)
    {
        if (addressDTO == null && addressId == null) return null;

        if (addressId != null) return GetById(addressId.Value);

        if(addressDTO != null) return Create(addressDTO);

        return null;
    }
```